### PR TITLE
Issue #000 chore: Added code to manage redirection if integrated with other extensions

### PR DESCRIPTION
**Pull request by praneettekdi**

### CODE REVIEW

File: src/components/com_tjcertificate/administrator/helpers/tjcertificate.php [hunk 1]
```diff
@@ -28,9 +28,16 @@ class TjCertificateHelper
 	 */
 	public static function addSubmenu($vName = '')
 	{
-		$layout = JFactory::getApplication()->input->get('layout', '', 'STRING');
+		$extension = JFactory::getApplication()->input->get('extension', '', 'STRING');
```
Comment (thite-amol) [inline src/components/com_tjcertificate/administrator/helpers/tjcertificate.php]: @praneettekdi Use the namespaces everywhere.

File: src/components/com_tjcertificate/administrator/includes/tjcertificate.php [hunk 2]
```diff
@@ -13,21 +13,37 @@
...
  * @since  1.0.0
  */
-class TjCertificateFactory
+class TJCERT
 {
+	/**
+	 * Holds the record of the loaded TJCertificate classes
+	 *
+	 * @var    array
+	 * @since  2.4.0
```
Comment (thite-amol) [inline src/components/com_tjcertificate/administrator/includes/tjcertificate.php]: @praneettekdi Use the tag to replace the versions

File: src/components/com_tjcertificate/administrator/views/template/tmpl/edit.php [hunk 3]
```diff
@@ -31,11 +31,18 @@
 $app = Factory::getApplication();
 $input = $app->input;
 
-$client = $input->getCmd('client', '');
+$client    = $input->getCmd('client', '');
+$extension = $input->getCmd('extension', '');
 
-if (!empty($client))
+$clientUrlAppend = '';
```
Comment (thite-amol) [inline src/components/com_tjcertificate/administrator/views/template/tmpl/edit.php]: @praneettekdi Avoid data manipulation in the view files

Comment (praneettekdi) [reply src/components/com_tjcertificate/administrator/views/template/tmpl/edit.php]: its fine if we manipulating URL for form submission

File: src/components/com_tjcertificate/administrator/includes/tjcertificate.php (hunk 2 above)
Comment (praneettekdi) [reply src/components/com_tjcertificate/administrator/includes/tjcertificate.php]: ok

File: src/components/com_tjcertificate/administrator/controllers/template.php [hunk 4]
```diff
@@ -41,9 +49,16 @@ public function __construct($config = array())
...
+		$jinput = $app->input;
+
+		if (empty($this->extension))
 		{
-			$this->client = Factory::getApplication()->input->get('tmplClient', '');
+			$this->extension = $jinput->get('extension', '');
+		}
+		elseif (empty($this->client))
+		{
+			$this->client = $jinput->get('tmplClient', '');
```
[the remark this answers is not in the document]
Comment (praneettekdi) [reply src/components/com_tjcertificate/administrator/controllers/template.php]: done

File: src/components/com_tjcertificate/administrator/libraries/certificate.php [hunk 5]
```diff
@@ -700,7 +682,7 @@ public function issueCertificate($replacements, $options)
 			}
 
 			// Get template details
-			$template = TjCertificateTemplate::getInstance($this->certificate_template_id);
+			$template = TJCERT::Template($this->certificate_template_id);
```
[the remark this answers is not in the document]
Comment (praneettekdi) [reply src/components/com_tjcertificate/administrator/libraries/certificate.php]: added additonal check for date validation for format 'Y-m-d'  and appended 23:59:59

File: src/components/com_tjcertificate/administrator/libraries/certificate.php [hunk 6]
```diff
@@ -723,7 +705,13 @@ public function issueCertificate($replacements, $options)
 
 			if (!empty($this->expired_on))
 			{
-				if (!(DateTime::createFromFormat('Y-m-d H:i:s', $this->expired_on) !== false))
+				// Check if only date is provided e.g. Y-m-d, then add 23:59:59
+				if (!(DateTime::createFromFormat('Y-m-d H:i:s', $this->expired_on) !== false)
+					&& (DateTime::createFromFormat('Y-m-d', $this->expired_on) !== false))
+				{
```
Comment (coolbung) [inline src/components/com_tjcertificate/administrator/libraries/certificate.php]: Consider changing this condition to `if ($this->expired_on != DateTime::createFromFormat('Y-m-d H:i:s', $this->expired_on))` ?

**Pull request merged**